assets/src/edit-story/utils/usePreventWindowUnload.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+
+const beforeUnloadListeners = new Map();
```

Comment:
This will work but with side-effects. IMHO it'd be a lot better to use `createContext()` here with the default value that will have what you want. Then you can use `useContext()` inside your `usePreventWindowUnload()`. You won't need any additional extra provider declarations - default context will work exactly like you need here. That's exactly what it's for.

E.g.

```
const PreventUnloadContext = createContext({listeners: new Map()});

function usePreventWindowUnload(...) {
  const context = useContext(PreventUnloadContext);
  const setPreventUnload = useCallback(() => {
    context.listeners.set(...);
     ...
  });
  return setPreventUnload;
}
```